django/db/migrations/autodetector.py

```diff
@@ -1307,19 +1309,22 @@ def _trim_to_apps(self, changes, app_labels):
         return changes
 
     @classmethod
-    def suggest_name(cls, ops):
+    def suggest_name(cls, migration):
```

Comment:
I would pass `migration.initial` instead of changing `ops` -> `migration`
```python
def suggest_name(cls, ops, initial=False):
```
It's unlikely that we will need anything else from `Migration` and it will reduce the number of changes.

Reply:
I had originally considered that, but in the end feel passing the `Migration` instance is the more elegant interface.

We need multiple pieces of information from the `Migration` instance. Rather than destructuring it down to its pieces, we can treat it as the rich data structure that it is and recognize that these pieces of information are already grouped together by a single reference. Destructuring the attributes loses this fidelity, IMO.

While minimal diffs are good and definitely provide value, the diff alone shouldn't decide what the preferred internal API is.

To help reduce the size of the diff, I broke the PR up into multiple commits. The first commit only changes the interface, so the effects are easily observable and isolated.

I made a few other changes to reduce the size of the diff:

- I updated `MigrationAutodetector.suggest_name()` to have the local variable `ops` so there are fewer lines changed within that method.

- I moved one of the new tests to an early commit. This tests passes on master but was added to catch unintentional regressions.

- I added an early commit to factor out the common code and string building in `MigrationAutodetector.arrange_for_graph()`.

With the above combined, the final feature change commit is now a single line outside of test code.

Reply:
Then I would move `suggest_name()` to the `Migration` class.

Reply:
That works for me. I've moved the method in the latest revision.